src/compositions/FieldsetHeader/FieldsetHeader.tsx

```diff
@@ -0,0 +1,105 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { IconSize } from "@elements/Icon/Icon";
+import { ReactComponent as AddSimple } from "@elements/Icon/Svg/AddSimple.svg";
+import { ReactComponent as CaretDown } from "@elements/Icon/Svg/CaretDown.svg";
+import { ReactComponent as CaretUp } from "@elements/Icon/Svg/CaretUp.svg";
+import { ReactComponent as Minus } from "@elements/Icon/Svg/Minus.svg";
+import Switch from "@elements/Switch/Switch";
+import { Size as SwitchSize } from "@utilities/enum";
+import generateRandomId from "@utilities/generateRandomId";
+import { merge } from "@utilities/merge";
+import { cloneElement, isValidElement, ReactElement, ReactNode } from "react";
+
+export enum Size {
+    Small = "Small",
+    Large = "Large",
+}
+
+export enum Type {
+    Default = "Default",
+    Switch = "Switch",
+    Accordion = "Accordion",
+    AddRemove = "AddRemove",
+}
+
+export type FieldsetHeaderProps = {
+    size?: Size;
+    active?: boolean;
+    decorator?: ReactNode;
+    type?: Type;
+    disabled?: boolean;
+    bold?: boolean;
+    heading: string;
+    onClick?: () => void;
+    as?: keyof JSX.IntrinsicElements;
+};
+
+const renderType = (type: Type, id: string, size: Size, active: boolean, disabled: boolean) => {
+    if (type === Type.Switch) {
```

Comment:
why not using switch/case?

Reply:
happy now? :smiling_face_with_three_hearts: